packages/widget/src/controllers/transfers/evm/build.ts

```diff
@@ -1,89 +1,114 @@
 import {
   EVMAssetTransfer,
-  FeeHandlerType,
-  type PercentageFee
+  FeeHandlerType
+} from '@buildwithsygma/sygma-sdk-core';
+import type {
+  Domain,
+  Environment,
+  EvmFee,
+  PercentageFee
 } from '@buildwithsygma/sygma-sdk-core';
 import { Web3Provider } from '@ethersproject/providers';
+import type { UnsignedTransaction, BigNumber } from 'ethers';
 import { constants, utils } from 'ethers';
-import { type FungibleTokenTransferController } from '../fungible-token-transfer';
+import type { EvmWallet } from 'packages/widget/src/context';
+
+type BuildEvmFungibleTransactionsArtifacts = {
+  pendingEvmApprovalTransactions: UnsignedTransaction[];
+  pendingTransferTransaction: UnsignedTransaction;
+  fee: EvmFee;
+  resourceAmount: BigNumber;
+};
 
 /**
  * @dev If we did proper validation this shouldn't throw.
  * Not sure how to handle if it throws :shrug:
  */
-export async function buildEvmFungibleTransactions(
-  this: FungibleTokenTransferController
-): Promise<void> {
-  //we already check that but this prevents those typescript errors
-  const provider = this.walletContext.value?.evmWallet?.provider;
-  const providerChaiId = this.walletContext.value?.evmWallet?.providerChainId;
-  const address = this.walletContext.value?.evmWallet?.address;
-  if (
-    !this.sourceNetwork ||
-    !this.destinationNetwork ||
-    !this.resourceAmount ||
-    !this.selectedResource ||
-    !this.destinationAddress ||
-    !provider ||
-    !address ||
-    providerChaiId !== this.sourceNetwork.chainId
-  ) {
-    this.estimatedGas = undefined;
-    this.resetFee();
-    return;
-  }
-
+export async function buildEvmFungibleTransactions({
+  evmWallet,
```

Comment:
you could've destructured here, but no need to update :)
```suggestion
  evmWallet: {
    address,
    provider,
    providerChainId
  },
```